services/src/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/FlowHelper.groovy

```diff
@@ -261,6 +262,14 @@ class FlowHelper {
         }
     }
 
+    List<FlowHistoryPayload> getFlowHistoryByAction(String flowId, String action) {
+        getFlowHistoryByAction(flowId, null, null, action)
+    }
+
+    List<FlowHistoryPayload> getFlowHistoryByAction(String flowId, Long timeFrom, Long timeTo, String action) {
+        northbound.getFlowHistory(flowId, timeFrom, timeTo).findAll { it.action == action }.last().histories
+    }
```

Comment:
I find this helper method to be too narrow. I 'd rather go for an explicit ` northbound.getFlowHistory(flowId, timeFrom, timeTo)` + search directly in test instead. Also, it should take `FlowAction` as fourth argument instead of `String`

Reply:
I like this better. Even more narrow, but fully hides pretty common operation for verifying final flow state. I would have made FlowState an argument to use the same method for up/down/inprogress. Also, I think it makes sense to make 'timeTo' to fall to `System.currentTimeMillis()` by default, since I don't see a popular situation where it will be something else. timeFrom is also a candidate for a default value, because most of the time I will not be interested in specifying it.